appveyor_after_build.ps1

```diff
@@ -0,0 +1,300 @@
+# Key for the project under test
+$sqProjectKey = "ProjectUnderTest";
+# Name of the quality profile used for analysis
+$testQualityProfileName = "ProfileForTest";
+
+#
+# Download and unzip SonarSqube to a well-known directory. If the directory exists, consider it as a cache. 
+#
+function FetchAndUnzipSonarQube
+{
+    $sqVersion = $env:SQ_Version;
+    $sqUrl = $sonarQubeDownloadUrlPrefix + $sqVersion + ".zip";    
+    $sqInstallPath = [System.IO.Path]::Combine($sonarQubeInstallPath, "sonarqube-" + $sqVersion)
+    
+    if ($LOCAL_DEBUG_RUN -and [System.IO.Directory]::Exists($sqInstallPath))
+    {
+        LogAppveyorMessage -Message "SonarQube $version is already present in $sqInstallPath"
+        
+    }
+    else
+    {
+        LogAppveyorMessage -Message "Downloading SonarQube from $sqUrl"
+    
+        if ([System.String]::IsNullOrWhiteSpace($sqUrl))
+        {
+            throw "The SonarQube server download URL was not specified. Make sure the yml file has SQ_URI pointing to this URL"
+        }    
+
+        $file = [System.IO.Path]::Combine($sonarQubeDownloadPath, "sonarqube-" + $sqVersion + ".zip")               
+        FetchAndUnzip $sqUrl $sonarQubeInstallPath $file $false
+    }
+
+    return $sqInstallPath;
+}
+
+
+#
+# Adds the C# plugin to the list of Appveyor artifacts
+#
+function PublishPatchedCsharpPlugin
+{    
+    if ($LOCAL_DEBUG_RUN)
+    {
+        return $localCSharpPluginPath;
+    }
+
+    $pluginPath = FindSingleFile ([System.IO.Path]::Combine($env:APPVEYOR_BUILD_FOLDER, "DeploymentArtifacts")) "*.jar"    
+    Push-AppveyorArtifact $pluginPath 
+
+    return $pluginPath;
+}
+
+#
+# Copy the C# plugin to the SQ plugins directory
+#
+function InstallCsPlugin
+{
+    param ([string]$sqServerPath, [string]$csPluginPath) 
+
+    if (![System.IO.File]::Exists($csPluginPath))
+    {
+        throw "Could not find the C# plugin at $csPluginPath"
+    }
+
+    $destination = [System.IO.Path]::Combine($sqServerPath, "extensions", "plugins", [System.IO.Path]::GetFileName($csPluginPath)); 
+    [System.IO.File]::Copy($csPluginPath, $destination, $true);
+}
+
+#
+# Start the SonarQube server as a process 
+#
+function StartSonarQubeServer
+{
+    param ([string]$sqServerPath)
+
+    StopSonarQubeServer
+
+    $sqStartBatPath = "$sqServerPath\bin\windows-x86-64\StartSonar.bat";
+    LogAppveyorMessage -Message "Starting the SonarQube server at $sqStartBatPath";  
+    
+    [void][System.Diagnostics.Process]::Start($sqStartBatPath);      
+}
+
+#
+# Stop the SQ server
+#
+function StopSonarQubeServer
+{
+    # Killing Java.exe is a reliable way of shutting down the SonarQube server
+    Get-Process | Where-Object {$_.Path -like "*java.exe"} | Stop-Process -Force
+    Start-Sleep -s 1
+}
+
+
+function WaitForSonarQubeToHaveStarted
+{
+    $command = { InvokeGetRestMethod "/api/projects/index?search=projectundertest"} 
+    Retry $command -maxRetries 30 -retryDelay 1 -Verbose
+}
```

Comment:
Sadly it takes ~3 minutes for the server to start for the first time on the Appveyor agents, making it the most expensive operation of all.